src/components/breakpointHandler.ts

```diff
@@ -0,0 +1,667 @@
+import * as vscode from 'vscode';
+import { DaCeVSCode } from '../extension';
+import { SdfgViewerProvider } from "./sdfgViewer"
+import fs = require("fs");
```

Comment:
If possible, I would try to migrate to the [VS Code API's version of the fs module](https://code.visualstudio.com/api/references/vscode-api#FileSystem). This would be beneficial because it works out of the box with remote VSCode instances and file systems (SSH, Docker, WSL, etc.). I know that you're using `isFileSync` and similar functions that are not part of the VSCode API, but there might be a way of doing the same with the aforementioned - see [this SO answer](https://stackoverflow.com/questions/58451856/vscode-api-check-if-path-exists/63372377#63372377). Let me know if this would work.

Reply:
Other than that it makes the code look messy, it does work, so that's fine for me. 
Let me know if there might be a more elegant way to catch the exceptions.
I had to make changes to when the extension saves the state:
- Old way: When the extension deactivates it calls the dispose function which saves the state in the .vscode folder
- New way: Every time we change the state (adding & removing Breakpoint Nodes and registering a function from DaCe) we save the state. The old way wouldn't work as it seems to cancel all Filesystem operations when the extension deactivates.

Reply:
That's quite sad, I wasn't aware that filesystem operations are aborted when `deactivate` is called..
I guess this saving at every change is a necessary evil then for now. Maybe there is a way to delegate it to 	 the [WorkspaceEdit API](https://code.visualstudio.com/api/references/vscode-api#WorkspaceEdit) and then save  that edit upon deactivation instead, however I am not sure if that would be much prettier. Potential premature optimization, so let's leave it for now and see how this behaves.